base/tps/src/org/dogtagpki/server/tps/channel/SecureChannelProtocol.java

```diff
@@ -0,0 +1,1528 @@
+package org.dogtagpki.server.tps.channel;
+
+import java.io.CharConversionException;
+import java.nio.ByteBuffer;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+import javax.crypto.BadPaddingException;
+
+import org.dogtagpki.tps.main.TPSBuffer;
+import org.mozilla.jss.CryptoManager;
+import org.mozilla.jss.NoSuchTokenException;
+import org.mozilla.jss.NotInitializedException;
+import org.mozilla.jss.crypto.Cipher;
+import org.mozilla.jss.crypto.CryptoToken;
+import org.mozilla.jss.crypto.EncryptionAlgorithm;
+import org.mozilla.jss.crypto.IVParameterSpec;
+import org.mozilla.jss.crypto.IllegalBlockSizeException;
+import org.mozilla.jss.crypto.KeyGenAlgorithm;
+import org.mozilla.jss.crypto.KeyGenerator;
+import org.mozilla.jss.crypto.KeyWrapAlgorithm;
+import org.mozilla.jss.crypto.KeyWrapper;
+import org.mozilla.jss.crypto.SymmetricKey;
+import org.mozilla.jss.crypto.SymmetricKeyDeriver;
+import org.mozilla.jss.crypto.TokenException;
+import org.mozilla.jss.pkcs11.PKCS11Constants;
+
+import com.netscape.certsrv.base.EBaseException;
+import com.netscape.cmscore.apps.CMS;
+import com.netscape.cmscore.apps.CMSEngine;
+import com.netscape.cmscore.security.JssSubsystem;
+import com.netscape.cmsutil.crypto.CryptoUtil;
+
+// It turns out that TPS only needs a small portion of the original TKS's SecureChannelProtocol class.
```

Comment:
The amount of code copied into this class is quite significant, so we need to be careful when modifying them in the future to make sure they are in-sync. Hopefully this is only a temporary workaround and we can remove the duplicate in the future.

Reply:
Yes, this is all temporary. There should be some other util based class that has this stuff in it. I wanted to keep the changes to a minimum, so I moved over the whole file and took out the stuff not needed on the TPS side.

Reply:
As for acheel's comment, I will take out the comment after merging the pr. Since it will come under the trivial checkin rule.